lib/jekyll/static_file.rb

```diff
@@ -1,6 +1,6 @@
 module Jekyll
   class StaticFile
-    attr_reader :relative_path, :extname
+    attr_reader :relative_path, :extname, :name
```

Comment:
This is a short way of defining:

```ruby
def name
  @name
end
```

Therefore, the best thing to do here is to set `@name` in the `initialize` method using the code you have on line 100 and then, when you are constructing `to_liquid` below, you can just call the `name` method instead of re-calculating its value. Does that make sense?